core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -1132,28 +1147,55 @@ private void reloadColumnAt(
-            // partitions. It is possible in theory, but this could be an algo for
-            // another day.
-            if (columnCountBits > this.columnCountBits) {
-                createNewColumnList(columnCount, pTransitionIndex, columnCountBits);
-            } else {
-                reshuffleColumns(columnCount, pTransitionIndex);
+        while (true) {
+            try {
+                long pTransitionIndex = metadata.createTransitionIndex(txnStructureVersion);
+                if (pTransitionIndex < 0) {
+                    if (configuration.getMicrosecondClock().getTicks() < deadline) {
+                        return false;
+                    }
+                    LOG.error().$("metadata read timeout [timeout=").$(configuration.getSpinLockTimeoutUs()).utf8("μs]").$();
+                    throw CairoException.instance(0).put("Metadata read timeout");
+                }
+                try {
+                    metadata.applyTransitionIndex(pTransitionIndex);
+                    final int columnCount = Unsafe.getUnsafe().getInt(pTransitionIndex + 4);
+
+                    int columnCountBits = getColumnBits(columnCount);
+                    // when a column is added we cannot easily reshuffle columns in-place
+                    // the reason is that we'd have to create gaps in columns list between
+                    // partitions. It is possible in theory, but this could be an algo for
+                    // another day.
+                    if (columnCountBits > this.columnCountBits) {
+                        createNewColumnList(columnCount, pTransitionIndex, columnCountBits);
+                    } else {
+                        reshuffleColumns(columnCount, pTransitionIndex);
+                    }
+                    // rearrange symbol map reader list
+                    reshuffleSymbolMapReaders(pTransitionIndex, columnCount);
+                    this.columnCount = columnCount;
+                    return true;
+                } finally {
+                    TableUtils.freeTransitionIndex(pTransitionIndex);
+                }
+            } catch (CairoException ex) {
+                // This is temporary solution until we can get multiple version of metadata not overwriting each other
+                if (isMetaFileMissingFileSystemError(ex)) {
+                    if (configuration.getMicrosecondClock().getTicks() < deadline) {
+                        LOG.info().$("error reloading metadata [table=").$(tableName)
+                                .$(", errno=").$(ex.getErrno())
+                                .$(", error=").$(ex.getFlyweightMessage()).I$();
+                        Os.pause();
+                    } else {
+                        LOG.error().$("metadata read timeout [timeout=").$(configuration.getSpinLockTimeoutUs()).utf8("μs]").$();
+                        throw CairoException.instance(0).put("Metadata read timeout");
```

Comment:
we are inside exception handling block, there is perhaps an opportunity not to hide the original exception?

Reply:
done in ed43dd5